eos-add-flatpak-apps-repos

```
@@ -151,6 +151,118 @@ def _update_deploy_file(deploy_file_path, idx, new_val, old_val=None):
+            if new_branch:
+                clashing_refs = _filter_refs(refs, name=name, kind=kind,
+                                             arch=arch, branch=new_branch)
+                if len(clashing_refs) > 0:
+                    logging.warning('Not migrating {}/{}/{}, new branch {} already exists'
+                                    .format(name, arch, branch, new_branch))
+                    continue
+
+                logging.info('Migrating {}/{}/{} to new branch {}'
+                             .format(name, arch, branch, new_branch))
+
+                # /var/lib/flatpak/${kind}/${name}/${arch}/${branch} , ${ref}-${subpaths}
+                (old_branch_dir, ref_dir) = os.path.split(deploy_dir)
+
+                # /var/lib/flatpak/${kind}/${name}/${arch}
+                arch_dir = os.path.dirname(old_branch_dir)
+
+                # /var/lib/flatpak/${kind}/${name}
+                name_dir = os.path.dirname(arch_dir)
+                current = os.path.join(name_dir, 'current')
+                current_tmp = current + '.tmp'
+
+                # /var/lib/flatpak/${kind}/${name}/${arch}/${new_branch}
+                new_branch_dir = os.path.join(arch_dir, new_branch)
+
+                # /var/lib/flatpak/${kind}/${name}/${arch}/${new_branch}/${ref}-${subpaths}
+                new_deploy_dir = os.path.join(new_branch_dir, ref_dir)
+
+                try:
+                    os.mkdir(new_branch_dir, mode=0o755)
+                    subprocess.check_call(['cp', '-al', deploy_dir, new_deploy_dir])
+                    os.symlink(ref_dir, os.path.join(new_branch_dir, 'active'))
+                    if kind == Flatpak.RefKind.APP:
+                        os.symlink('{}/{}'.format(arch, new_branch), current_tmp)
+                        os.rename(current_tmp, current)
+                    # this is unsafe whilst flatpaks are running, but: this is a startup job
+                    shutil.rmtree(old_branch_dir, ignore_errors=True)
```

I'd probably add these two lines above to an `else` block because, even though we tell it to ignore errors, if there's a problem after the `os.rename` call, we'd end up with a broken app AFAICS. i.e. after we have all the links in place, we don't want to risk it removing the dir tree (and a simple unnoticed, last minute addition syntax error above would trigger that).

---

fixed in new PR (along with a bug about clearing up current_tmp)